docs/proposals/clusters.md

```diff
@@ -0,0 +1,214 @@
+
+To eliminate these issues, this proposal outlines the redesign of the current data model to allow for the grouping of hosts within a single site, and provides a configuration layout to allow host groups to utilize different configuration versions.
+
+# Quick Definitions
+
+For clarity.
+
+Configuration: all workflows, boots, OSes, assets defined in the data model. Essentially every document in the current data model not under the `sites/` directory.
+
+Inventory: all hosts defined in a particular site. Currently gained from all files under `sites/` that are not `env.yml`.
+
+Site: The highest level of organization for metadata/inventory. Usually refers to a single physical deployment of hosts.
+
+Cluster: A subset of site hosts that runs under a specific configuration and does not have depedencies with other hosts outside of it's subset.
+
+
+# Host Clusters
+
+## Requirements
+
+- Logically separate hosts into clusters within a single site
+- Groups are independent w.r.t. rollout policy/workflow dependency
+
+## Design
+
+The current logical layout for a data model is:
+
+```
+├── assets
+│   └── Config files, templates
+├── boot
+│   └── Pair configs + OSes, and metadata
+├── os
+│   └── OS kernel, initrd, cmdline information
+├── sites
+│   └── site-name
+│       ├── env.yml <- Specifies agent configurations, site-wide metadata, subnet information
+│       └── inventory.yml <- Host definitions deliminated as individual YAML documents 
+└── workflows
+    └── Chain multiple boots together, inter-workflow dependencies and rollout policy
+```
+
+In this layout, all hosts are declared in a homogenous pool, a site's "inventory". An example `inventory.yml`: 
+
+```
+---
+name: host1
+interfaces:
+  - type: physical
+    subnet: subnet1
+    mac: 00:00:00:00:00:01
+    ipv4: 127.0.0.1
+metadata:
+    mstr: etcd1
+workflow: coreos-cloudconfig
+---
+name: host2
+interfaces:
+  - type: physical
+    subnet: vagrant1
+    bmc:
+      type: ssh
+      username: core
+      keypath: /secrets/id_rsa
+    mac: 00:00:00:00:00:02
+    ipv4: 127.0.0.2
+metadata:
+    mstr: etcd2
+workflow: coreos-ignition
+---
+name: host3
+interfaces:
+  - type: physical
+    subnet: vagrant1
+    mac: 00:00:00:00:00:03
+    ipv4: 127.0.0.3
+metadata:
+    mstr: etcd3
+    random: data
+workflow: something-else
+```
+
+To accomodate logically separated pools of hosts, multiple inventory files will be specified, with each inventory file providing the hosts that belong to a single cluster in the same format shown above:
+
+```
+├── sites
+│   └── site-name
+│       ├── env.yml <- Specifies agent configurations, site-wide metadata, subnet information
```

Comment:
The parsing logic in [parse.go](/CiscoCloud/vaquero/blob/680116a/model/parse.go#L11)'s `collectSite` first reads the `env.yml` file, and then reads inventory from the remainder of the files. This would be a logical place to potentially branch the logic for parsing the existing model (hereafter: `modelv1`) versus the new model (`modelv2`).

Please considering adding a `version:` key to the `env.yml` to enable an explicit identifier that this is where the behavior should change. (Alternatively: create an additional file, such as `version.yml`, for this purpose. I think putting it in `env.yml` should be sufficient.)